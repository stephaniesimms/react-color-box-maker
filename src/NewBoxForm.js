import React, { Component } from 'react';
import uuid from 'uuid/v4';

class NewBoxForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      backgroundColor: "",
      width: "",
      height: ""
    };
    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  handleChange(evt) {
    this.setState({
      [evt.target.name]: evt.target.value
    });
  }

  handleSubmit(evt) {
    evt.preventDefault();
    this.props.makeBox({ ...this.state, id: uuid() });
    this.setState({
      backgroundColor: "",
      height: "",
      width: ""
    });
  }

  render() {
    return (
      <div>
        <form onSubmit={this.handleSubmit}>

          <label htmlFor="backgroundColor">Background Color:</label>
          <input id="backgroundColor" name="backgroundColor"
            value={this.state.backgroundColor}
            onChange={this.handleChange}></input>

          <label htmlFor="width">Width:</label>
          <input id="width" name="width"
            value={this.state.width}
            onChange={this.handleChange}></input>

          <label htmlFor="height">Height:</label>
          <input id="height" name="height"
            value={this.state.height}
            onChange={this.handleChange}></input>

          <button>Submit</button>

        </form>
      </div>
    );
  }
}

export default NewBoxForm;