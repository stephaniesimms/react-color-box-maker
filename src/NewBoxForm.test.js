import React from 'react';
import ReactDOM from 'react-dom';
import toJson from "enzyme-to-json";
import { mount } from 'enzyme';
import NewBoxForm from './NewBoxForm';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<NewBoxForm />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("matches snapshot", function () {
  let wrapper = mount(<NewBoxForm />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("allows for changes in backgroundColor, width, height", function () {
  let wrapper = mount(<NewBoxForm />);
  
  const backColorInput = wrapper.find("#backgroundColor")
  backColorInput.instance().value = "magenta"
  backColorInput.simulate("change")

  expect(wrapper.state().backgroundColor).toEqual("magenta");

  const widthInput = wrapper.find("#width");
  widthInput.instance().value = '20';
  widthInput.simulate("change");

  expect(wrapper.state().width).toEqual('20');

  const heightInput = wrapper.find("#height");
  heightInput.instance().value = '25';
  heightInput.simulate("change");

  expect(wrapper.state().height).toEqual('25');
});

it("runs a mocked fn on submit", function () {
  const submitFn = jest.fn();
  let wrapper = mount(
    <NewBoxForm makeBox={submitFn} />
  );
  const form = wrapper.find("form")

  form.simulate("submit")

  expect(submitFn).toHaveBeenCalled();
});

// duplicates test above for confirming changes to inputs
it("resets state on submit", function () {
  const submitFn = jest.fn();
  let wrapper = mount(<NewBoxForm makeBox={submitFn} />);
  
  const backColorInput = wrapper.find("#backgroundColor")
  backColorInput.instance().value = "magenta"
  backColorInput.simulate("change")

  expect(wrapper.state().backgroundColor).toEqual("magenta");

  const widthInput = wrapper.find("#width");
  widthInput.instance().value = '20';
  widthInput.simulate("change");

  expect(wrapper.state().width).toEqual('20');

  const heightInput = wrapper.find("#height");
  heightInput.instance().value = '25';
  heightInput.simulate("change");

  expect(wrapper.state().height).toEqual('25');

  const form = wrapper.find("form");
  form.simulate("submit");

  // after submit, we expect the state to reset
  expect(wrapper.state()).toEqual({
    backgroundColor: '',
    height: '',
    width: ''
  })
});

