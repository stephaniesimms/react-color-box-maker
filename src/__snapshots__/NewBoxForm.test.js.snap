// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<NewBoxForm>
  <div>
    <form
      onSubmit={[Function]}
    >
      <label
        htmlFor="backgroundColor"
      >
        Background Color:
      </label>
      <input
        id="backgroundColor"
        name="backgroundColor"
        onChange={[Function]}
        value=""
      />
      <label
        htmlFor="width"
      >
        Width:
      </label>
      <input
        id="width"
        name="width"
        onChange={[Function]}
        value=""
      />
      <label
        htmlFor="height"
      >
        Height:
      </label>
      <input
        id="height"
        name="height"
        onChange={[Function]}
        value=""
      />
      <button>
        Submit
      </button>
    </form>
  </div>
</NewBoxForm>
`;
